tests/Unit/HtmlProcessor/AbstractHtmlProcessorTest.php

```diff
@@ -732,4 +746,42 @@ public function getDomDocumentVoidElementNotHasChildNodes(string $htmlWithNonXml
             self::assertFalse($element->hasChildNodes());
         }
     }
+
+    /**
+     * Asserts that two HTML strings are equal, allowing for whitespace differences in the HTML element itself (but not
+     * its descendants) and after its closing tag.
+     *
+     * @param string $expected
+     * @param string $actual
+     * @param string $message
+     */
+    private static function assertEqualsHtml(string $expected, string $actual, string $message = '')
+    {
+        $normalizedExpected = self::normalizeHtml($expected);
+        $normalizedActual = self::normalizeHtml($actual);
+
+        self::assertSame($normalizedExpected, $normalizedActual, $message);
+    }
+
+    /**
+     * Normalizes whitespace in the HTML element itself (but not its descendants) and after its closing tag, with a
+     * single newline inserted or replacing whitespace at positions where whitespace may occur but is superfluous.
+     *
+     * @param string $html
+     *
+     * @return string
+     */
+    private static function normalizeHtml(string $html)
```

Comment:
This method can have a `string` return type annotation.

Reply:
Good catch.  I've added #857 to see if missing return type anntations could be automatically caught.